File: Java/프로그래머스/1/12982. 예산/예산.java
import java.util.*;
class Solution {
    public int solution(int[] d, int budget) {
        /*
        1.d를 돌면서 i들의 합이 9를 초과하기 전까지 count를 해준다. 
        2.count 값을 리턴한다.
        이때 배열 d는 오름차순정렬을 해줘야, 최대를 구할 수 있다. 
        */
        
        Arrays.sort(d);
        
        int sum = 0;
        int count = 0; 
        
        for(int i = 0; i < d.length; i++){
            sum = sum + d[i]; 
            
            if(sum > budget){
                return count; 
            }
            count++;
        }

        return count;
    }
}

File: Python3/프로그래머스/1/12982. 예산/예산.py
def solution(d, budget):
    cnt = 0
    d_sort = sorted(d) #오름차순으로 정렬한다.[1,2,3,4,5]
    for i in d_sort: 
        budget -= i 
        if budget < 0: 
            break 
        cnt += 1 
    return cnt